pkg/resources/mutation/cluster/provisioningcluster.go

```diff
@@ -187,14 +190,16 @@ func (m *ProvisioningClusterMutator) handlePSACT(request *admission.Request, clu
 				secretAnnotation: cluster.Name,
 			}
 			data := map[string][]byte{
-				secretKey: fileContent,
+				SecretKey: fileContent,
 			}
 			err = m.ensureSecret(cluster.Namespace, secretName, data, anno)
 			if err != nil {
 				return nil, fmt.Errorf("[provisioning cluster mutator] failed to create or update the secret for the admission configuration file: %w", err)
 			}
-			// set relevant fields in the cluster
-			addMachineSelectorFile(MachineSelectorFileForPSA(secretName, mountPath), cluster)
+			// drop then set relevant fields if they exist in the cluster
+			dropMachineSelectorFile(MachineSelectorFileForPSA(secretName, mountPath, ""), cluster, true)
+			hash := sha256.Sum256(fileContent)
```

Comment:
Nit: sha256 is fine here but for future uses, if you are only doing a checksum and nothing security related md5 is a faster hashing algorithm.

Reply:
Good to know that 😄  
SHA256 is used here because it is commonly used in rancher. 
I will keep it because changes need to be done both here and in rancher/rancher.

Reply:
On the hash algorithmn selection, we just need to be mindful of the implications on users that use Rancher in regulated environments. Ideally sticking to NIST approved options (e.g. `sha256`, `sha512`).

But looking on the bright side, `SHA-NI` [improvements](https://go-review.googlesource.com/c/go/+/408795/1) are coming for Go's `sha256` implementation, which may even mean it beat `md5`'s performance. :smile: